src/cloudai/configurator/gymnasium_adapter.py

```
@@ -0,0 +1,165 @@
+
+        Raises:
+            ValueError: if ``params`` does not cover exactly the tunable + fixed param keys.
+        """
+        self._assert_keys(params.keys(), set(self._tunable_params) | set(self._fixed_params), "raw params")
+        return self._step_with_params(params)
+
+    def render(self) -> None:
+        self._env.render()
+
+    @staticmethod
+    def _assert_keys(received: Any, expected: set[str], ctx: str) -> None:
```

Ruff (0.15.15)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `received`

(ANN401)